### PR TITLE
Prototype of clevis integration from CLI

### PR DESCRIPTION
**Pull request by jbaublitz**
Closes stratis-storage/project#192

### CODE REVIEW

**Comment by jbaublitz** [comment]
Blocked by stratis-storage/stratisd#2253

File: src/stratis_cli/_actions/_top.py [hunk 1]
```diff
@@ -795,7 +795,68 @@ def list_keys(_):
...
+        (changed, return_code, return_msg) = Pool.Methods.BindClevis(
+            get_object(pool_object_path),
+            {
+                "key_desc": namespace.key_desc,
+                "tang_url": namespace.tang_url,
+                "tang_thp": namespace.tang_thp,
+            },
+        )
+
+        if return_code != 0:
```
Comment (mulkieran) [inline src/stratis_cli/_actions/_top.py:824]: We're using StratisdErrors enumerated type now.

File: src/stratis_cli/_actions/_top.py [hunk 2]
```diff
@@ -795,7 +795,68 @@ def list_keys(_):
...
+        (pool_object_path, _) = next(
+            pools(props={"Name": pool_name})
+            .require_unique_match(True)
+            .search(managed_objects)
+        )
+        (changed, return_code, return_msg) = Pool.Methods.UnbindClevis(
+            get_object(pool_object_path), {}
+        )
+
+        if return_code != 0:
```
Comment (mulkieran) [inline src/stratis_cli/_actions/_top.py:852]: Same as above.

File: src/stratis_cli/_parser/_pool.py [hunk 3]
```diff
@@ -139,10 +139,50 @@
...
+    ),
     (
         "unlock",
         dict(
             help="Unlock all of the currently locked encrypted pools",
+            args=[
+                (
+                    "unlock_method",
+                    dict(
+                        default="keyring",
```
Comment (mulkieran) [inline src/stratis_cli/_parser/_pool.py:180]: You can add "choices" field here. What should the second choice be? Just "clevis"?

File: src/stratis_cli/_parser/_pool.py [hunk 4]
```diff
@@ -139,10 +139,50 @@
...
+                    ),
+                ),
+            ],
+            func=TopActions.bind_clevis,
+        ),
+    ),
+    (
+        "unbind-clevis",
+        dict(
+            help="Unlock all of the currently locked encrypted pools",
```
Comment (mulkieran) [inline src/stratis_cli/_parser/_pool.py:167]: This help text here looks like a copy-paste error.

File: src/stratis_cli/_parser/_pool.py [hunk 5]
```diff
@@ -139,10 +139,50 @@
             func=TopActions.add_cache_devices,
         ),
     ),
+    (
+        "bind-clevis",
+        dict(
+            help="Unlock all of the currently locked encrypted pools",
```
Comment (mulkieran) [inline src/stratis_cli/_parser/_pool.py:145]: help text looks wrong here, too.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Code all makes sense, some minor requests. It won't hurt to update the man pages at this time, especially with information about the thumbprint, since the user has to supply it.

File: src/stratis_cli/_parser/_pool.py (hunk 3 above)
Comment (jbaublitz) [reply src/stratis_cli/_parser/_pool.py:180]: If we change it to something else, that'll need to be updated in stratisd too.

**Comment by mulkieran** [comment]
Pushing this back to in progress as we had considerable discussions in standup and suggestions about modifying both the CLI and the D-Bus API.

**Comment by jbaublitz** [comment]
That sounds good to me!

**Pull request closed without merging**